packages/caravan-fees/src/types.ts

```diff
@@ -0,0 +1,444 @@
+   */
+  dustThreshold?: Satoshis;
+
+  /**
+   * The Bitcoin network to use (mainnet, testnet, or regtest).
+   */
+  network: Network;
+
+  /**
+   * The type of script used for the transaction (e.g., "p2pkh", "p2sh", "p2wpkh", "p2wsh").
+   * This affects how the transaction is constructed and signed.
+   */
+  scriptType: string;
+
+  /**
+   * Optional array of input templates to use in the transaction.
+   */
+  inputs?: BtcTxInputTemplate[];
+
+  /**
+   * Optional array of output templates to use in the transaction.
+   */
+  outputs?: BtcTxOutputTemplate[];
+
+  /**
+   * The number of signatures required in a multisig setup.
+   * This is used for multisig transactions and affects the transaction size.
+   */
+  requiredSigners?: number;
+
+  /**
+   * The total number of signers in a multisig setup.
+   * This is used along with requiredSigners for multisig transactions.
+   */
+  totalSigners?: number;
+}
+
+export enum ScriptType {
+  P2PKH = "p2pkh",
+  P2SH = "p2sh",
+  P2WPKH = "p2wpkh",
+  P2WSH = "p2wsh",
+  P2SH_P2WPKH = "p2sh-p2wpkh",
+  P2SH_P2WSH = "p2sh-p2wsh",
+  UNKNOWN = "unknown",
+}
+
+/**
+ * Options for creating a cancel RBF transaction.
+ */
+export interface CancelRbfOptions {
+  /**
+   * The hex-encoded original transaction to be replaced.
+   */
+  originalTx: string;
+
+  /**
+   * Array of available UTXOs, including the original transaction's inputs.
+   */
+  availableInputs: UTXO[];
+
+  /**
+   * The address where all funds will be sent in the cancellation transaction.
+   */
+  cancelAddress: string;
+
+  /**
+   * The Bitcoin network being used (e.g., mainnet, testnet).
+   */
+  network: Network;
+
+  /**
+   * The dust threshold in satoshis. Outputs below this value are considered "dust"
+   * and may not be economically viable to spend.
+   */
+  dustThreshold: Satoshis;
+
+  /**
+   * The type of script used for the transaction (e.g., P2PKH, P2SH, P2WSH).
+   */
+  scriptType: string;
+
+  /**
+   * The number of required signers for the multisig setup.
+   */
+  requiredSigners: number;
+
+  /**
+   * The total number of signers in the multisig setup.
+   */
+  totalSigners: number;
+
+  /**
+   * The target fee rate in satoshis per virtual byte. This is used to calculate
+   * the appropriate fee for the transaction.
+   */
+  targetFeeRate: FeeRateSatsPerVByte;
+
+  /**
+   * The absolute fee of the original transaction in satoshis.
+   */
+  absoluteFee: Satoshis;
+
+  /**
+   * Whether to attempt full RBF even if the transaction doesn't signal it.
+   * @default false
+   */
+  fullRBF?: boolean;
+
+  /**
+   * If true, enforces stricter validation rules.
+   * @default false
+   */
+  strict?: boolean;
+}
+
+/**
+ * Options for creating an accelerated RBF transaction.
+ */
+export interface AcceleratedRbfOptions
+  extends Omit<CancelRbfOptions, "cancelAddress"> {
+  /**
+   * The index of the change output in the original transaction.
+   */
+  changeIndex?: number;
```

Comment:
do you support both changeIndex and changeAddress? or will one override the other? Should probably document (or throw an error if both are provided and document that, which is probably better just for predictability)

Reply:
added details regarding this now